### ISSUE TITLE
Multiple GPU support

### ISSUE THREAD

**Issue by HYPJUDY** [jayleicn/TVQA#1]
Does the code support multiple GPU training?

**Comment by jayleicn** [comment]
Nope, I did not write that part of the code.

**Comment by HYPJUDY** [comment]
Since each epoch training spends about 1h45min with one gpu in my environment, I want to save time by training with multiple gpus. Simply add `model = nn.DataParallel(model)` in L121 of the following code snippet can support multiple GPU training.
https://github.com/jayleicn/TVQA/blob/21cc7e3b549e616c51bd4d3953eed338a75d3939/main.py#L116-L124
By default, all available gpus will be used. Use command like `CUDA_VISIBLE_DEVICES=0,1,2 python main.py --input_streams sub` to specify specific gpu(s).
By doing this (without changing any other configuration like batch size), gpu memory is distributed to multiple gpus but **the training time for each epoch doesn't decrease** as expected. Do you know why and how to solve it?
BTW, the accuracy also doesn't change.

Ref: [DATA PARALLELISM](https://pytorch.org/tutorials/beginner/blitz/data_parallel_tutorial.html) and [MULTI-GPU EXAMPLES](https://pytorch.org/tutorials/beginner/former_torchies/parallelism_tutorial.html)

**Comment by jayleicn** [comment]
Most likely the overhead of the code is on I/O or CPU.
